Add BREAKING.md updates from recent changes (#7314)
microsoft/FluidFramework: BREAKING.md

- [Property removed from IFluidDataStoreContext](#Property-removed-from-IFluidDataStoreContext)
- [Changes to IFluidDataStoreFactory](#Changes-to-IFluidDataStoreFactory)
- [FlushMode enum values renamed](#FlushMode-enum-values-renamed)
- [name removed from ContainerSchema](#name-removed-from-ContainerSchema)
- [Anonymous return types for container calls in client packages](#Anonymous-return-types-for-container-calls-in-client-packages)
- [createContainer and getContainer response objects properties renamed](#createContainer-and-getContainer-response-objects-properties-renamed)
- [tinylicious and azure clients createContainer now detached](#tinylicious-and-azure-clients-createContainer-now-detached)

> - [container id is returned from new attach() and not exposed on the container](#container-id-is-returned-from-new-attach()-and-not-exposed-on-the-container)

Reviewer: nit: this link doesn't seem to work, probably the `()` needs to be removed?

Author: Hmm interesting. I actually checked this and it works in vscode. Removing them worked too so I did that but yea. Nice catch :)

- `FlushMode.Manual` to `FlushMode.TurnBased`
- `FlushMode.Automatic` to `FlushMode.Immediate`

### `name` removed from ContainerSchema

> The `name` property on the ContainerSchema was used for multi-container scenarios but has not materialized to be a useful schema property. The feedback has been negative to neutral and the goal is to remove it before it becomes formalized. Support for multi-container scenarios, if any is required, will be addressed as a future change.

Reviewer: ```suggestion
The `name` property on the ContainerSchema was used for multi-container scenarios but has not materialized to be a useful schema property. The feedback has been negative to neutral so it is being removed before it becomes formalized. Support for multi-container scenarios, if any is required, will be addressed as a future change.
```

Reviewer: Actually this is the same note as `### TinyliciousClient and AzureClient container API changed` below right?  So it's not a 0.47 change?

```javascript
// `@fluidframework/azure-client`
createContainer(containerSchema: ContainerSchema): Promise<{
    container: FluidContainer;
    services: AzureContainerServices;
}>;
getContainer(id: string, containerSchema: ContainerSchema): Promise<{
    container: FluidContainer;
    services: AzureContainerServices;
}>;

// `@fluidframework/tinylicious-client`
createContainer(containerSchema: ContainerSchema): Promise<{
    container: FluidContainer;
    services: TinyliciousContainerServices;
}>;
getContainer(id: string, containerSchema: ContainerSchema): Promise<{
    container: FluidContainer;
    services: TinyliciousContainerServices;
}>;
```

### createContainer and getContainer response objects properties renamed
For all `*-client` packages `createContainer` and `getContainer` would return an object with `fluidContainer` and `containerServices`. These have been renamed to the following for brevity.

- fluidContainer => container
- containerServices => services

```javascript

> // old
> const { fluidContainer, containerServices } = client.getContainer;
> 
> // new
> const { container, services } = client.getContainer;

Reviewer: ```suggestion
// old
const { fluidContainer, containerServices } = client.getContainer(...);

// new
const { container, services } = client.getContainer(...);
```
nitty nitty
